rust/src/datatypes.rs

```diff
@@ -87,8 +235,8 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_define_schema() {
-        let _person = Schema::new(vec![
+    fn create_struct_type() {
+        let _person = DataType::Struct(vec![
```

Comment:
@andygrove are we still using Schema? The tests now all reference Vecs of fields in a single DataType.

Reply:
I think Schema is becoming redundant. I'm still  using it in my project to define a `RecordBatch` but I could switch to using a `Vec<Field>` instead. Do you want to file a JIRA for removing Schema?

Reply:
OK, though to what extent should we mirror https://arrow.apache.org/docs/metadata.html? 

That would make a `Schema` of `Field`s.
`Struct(Vec<Field>)` would go from a variant of `DataType` to defined as `children` on a `Field`, whose type was only `Struct`?

(tbc these are low confidence and I'm only recently learning, so shoot me down if I'm off)